### PR TITLE
Assign duration support

### LINKED ISSUE #105: Set appointment duration during creation
As a nurse, I am able to set the duration of the appointment so that I can adjust the end time of the appointment

### PR DESCRIPTION
**Pull request by WadeForward**
Add duration support for AssignCommand and AssignCommandParser.
Update the related tests and User Guide.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -159,18 +159,19 @@ Examples:
...
 
-Format: `assign INDEX d/DATE t/TIME`
+Format: `assign INDEX d/DATE t/TIME [dur/DURATION]`
 
 * Puts the patient at the specified INDEX into an appointment time slot.
 * The INDEX refers to the index number indicated in the patient list.
 * The INDEX **must be a positive integer** 1, 2, 3, …​
 * The `DATE` and `TIME` of the appointment must be included.
-* The timeslot indicated by `DATE` and `TIME` must be available.
+* The `DURATION` is in unit of minute, with a default value of `60` when the term is omitted.
```
Comment (JinHao-L) [inline docs/UserGuide.md]: The phrasing seems a bit off.  
Would it be better if we phrase it as `The 'DURATION' is measured in minutes and will be defaulted to 60 minutes if omitted`?

Do add on if the phrasing can be improved further. 👍

**Review by JinHao-L — APPROVED** [review]
Other than the comment in the userguide, LGTM! 👍

File: docs/UserGuide.md (hunk 1 above)
Comment (Avalionnet) [reply docs/UserGuide.md]: Jin Hao's phrasing looks good to me too 👍🏻

File: docs/UserGuide.md [hunk 2]
```diff
@@ -159,18 +159,19 @@ Examples:
...
+* The time slot indicated by `DATE` and `TIME` must be available.
 * The specified `DATE` and `TIME` must be in the future.
 
 Examples:
-* `assign 1 d/Sunday t/2am` books an appointment at the upcoming Sunday, 2am for the 1st patient in the list.
-* `assign 3 d/02-03-2021 t/1130` books an appointment on 02/03/2021, 11:30am for the 3rd patient in the list.
+* `assign 1 d/Sunday t/2am dur/40` books an appointment of 40 minutes on the upcoming Sunday, 2am for the 1st patient in the list.
```
Comment (Avalionnet) [inline docs/UserGuide.md:173]: Hmm, perhaps you can consider the following phrase instead, "books an appointment at 2am for the 1st patient in the list on the coming Sunday for a duration of 40 minutes.

Additionally, I think it'll be better if we use 2pm as an example instead cause 2am is in the middle of the night and outside the usual operating hours 😅

File: docs/UserGuide.md [hunk 3]
```diff
@@ -159,18 +159,19 @@ Examples:
...
 * The INDEX refers to the index number indicated in the patient list.
 * The INDEX **must be a positive integer** 1, 2, 3, …​
 * The `DATE` and `TIME` of the appointment must be included.
-* The timeslot indicated by `DATE` and `TIME` must be available.
+* The `DURATION` is in unit of minute, with a default value of `60` when the term is omitted.
+* The time slot indicated by `DATE` and `TIME` must be available.
 * The specified `DATE` and `TIME` must be in the future.
 
 Examples:
-* `assign 1 d/Sunday t/2am` books an appointment at the upcoming Sunday, 2am for the 1st patient in the list.
```
Comment (Avalionnet) [inline docs/UserGuide.md:172]: Same here for the am!

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 4]
```diff
@@ -28,31 +31,33 @@
             + "Parameters: "
             + "PATIENT INDEX (must be a positive integer) "
             + PREFIX_DATE + "DATE "
-            + PREFIX_TIME + "TIME \n"
+            + PREFIX_TIME + "TIME "
+            + "[" + PREFIX_DURATION + "DURATION] (minute in unit) \n"
```
Comment (Avalionnet) [inline src/main/java/seedu/address/logic/commands/AssignCommand.java:35]: Would it be better if we phrase it as "(unit in minutes)" instead?

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 5]
```diff
@@ -80,22 +85,88 @@ public CommandResult execute(Model model) throws CommandException {
...
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof AssignCommand // instanceof handles nulls
-                && targetIndex.equals(((AssignCommand) other).targetIndex)); // state check
+                && targetIndex.equals(((AssignCommand) other).targetIndex)
+                && durationSupporter.equals(((AssignCommand) other).durationSupporter)); // state check
     }
 
+    public static class DurationSupporter extends DateTimeLoader {
```
Comment (Avalionnet) [inline src/main/java/seedu/address/logic/commands/AssignCommand.java:110]: Excellent work on this 👍🏻

File: docs/UserGuide.md (hunk 1 above)
Comment (WadeForward) [reply docs/UserGuide.md]: Thanks guys!
Yes that is good indeed, I have changed it.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 6]
```diff
@@ -176,4 +179,26 @@ public static Remark parseRemark(String remark) throws ParseException {
         }
         return new Remark(trimmedRemark);
     }
+
+    /**
```
Comment (Avalionnet) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:183]: Good job considering all the cases 👍🏻

**Review by Avalionnet — APPROVED** [review]
Excellent work and commendable effort 👍🏻 LGTM other than the few minor comments 😄

File: docs/UserGuide.md (hunk 2 above)
Comment (WadeForward) [reply docs/UserGuide.md:173]: Good spot and thank for pointing them out!

File: src/main/java/seedu/address/logic/commands/AssignCommand.java (hunk 4 above)
Comment (WadeForward) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java:35]: Definitely!
And I just realized that use `in minutes` to express it 🤣 
I think that is simpler and clearer?

File: src/main/java/seedu/address/logic/commands/AssignCommand.java (hunk 5 above)
Comment (WadeForward) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java:110]: Haha, you too!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 6 above)
Comment (WadeForward) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:183]: Good job reviewing all my codes 👍

**Pull request merged**